metrics_sdk/test/opentelemetry/sdk/metrics/aggregation/histogram_mapping_test.rb

```diff
@@ -0,0 +1,356 @@
+      _(exponent_mapping.map_to_index(2**-1040)).must_equal(-64)
+      _(exponent_mapping.map_to_index(2**-1025)).must_equal(-64)
+      _(exponent_mapping.map_to_index(2**-1024)).must_equal(-64)
+      _(exponent_mapping.map_to_index(2**-1023)).must_equal(-64)
+      _(exponent_mapping.map_to_index(2**-1022)).must_equal(-64)
+      _(exponent_mapping.map_to_index(2**-1009)).must_equal(-64)
+      _(exponent_mapping.map_to_index(2**-1008)).must_equal(-64)
+      _(exponent_mapping.map_to_index(2**-1007)).must_equal(-63)
+      _(exponent_mapping.map_to_index(2**-993)).must_equal(-63)
+      _(exponent_mapping.map_to_index(2**-992)).must_equal(-63)
+      _(exponent_mapping.map_to_index(2**-991)).must_equal(-62)
+      _(exponent_mapping.map_to_index(2**-977)).must_equal(-62)
+      _(exponent_mapping.map_to_index(2**-976)).must_equal(-62)
+      _(exponent_mapping.map_to_index(2**-975)).must_equal(-61)
+    end
+
+    it 'test_exponent_mapping_min_scale_negative_10' do
+      exponent_mapping = OpenTelemetry::SDK::Metrics::Aggregation::ExponentialHistogram::ExponentMapping.new(exponent_mapping_min_scale)
+      _(exponent_mapping.map_to_index(1.000001)).must_equal(0)
+      _(exponent_mapping.map_to_index(1)).must_equal(-1)
+      _(exponent_mapping.map_to_index(Float::MAX)).must_equal(0)
+      _(exponent_mapping.map_to_index(Float::MIN)).must_equal(-1)
+    end
+
+    it 'test_exponent_index_max' do
+      (-10...0).each do |scale|
+        exponent_mapping = OpenTelemetry::SDK::Metrics::Aggregation::ExponentialHistogram::ExponentMapping.new(scale)
+
+        inds = exponent_mapping.map_to_index(MAX_NORMAL_VALUE)
+        max_index = ((MAX_NORMAL_EXPONENT + 1) >> -scale) - 1
+
+        _(inds).must_equal(max_index)
+
+        boundary = exponent_mapping.get_lower_boundary(inds)
+        _(boundary).must_equal(right_boundary(scale, max_index))
+
+        error = assert_raises(StandardError) do
+          exponent_mapping.get_lower_boundary(inds + 1)
+        end
+        assert_equal('mapping underflow', error.message)
+      end
+    end
+
+    it 'test_exponent_index_min' do
+      (-10..0).each do |scale|
+        exponent_mapping = OpenTelemetry::SDK::Metrics::Aggregation::ExponentialHistogram::ExponentMapping.new(scale)
+
+        min_index = exponent_mapping.map_to_index(MIN_NORMAL_VALUE)
+        boundary = exponent_mapping.get_lower_boundary(min_index)
+
+        correct_min_index = MIN_NORMAL_EXPONENT >> -scale
+
+        correct_min_index -= 1 if MIN_NORMAL_EXPONENT % (1 << -scale) == 0
+
+        # We do not check for correct_min_index to be greater than the
+        # smallest integer because the smallest integer in Ruby is -Float::INFINITY.
+
+        _(correct_min_index).must_equal(min_index)
+
+        correct_boundary = right_boundary(scale, correct_min_index)
+
+        # truffleruby will fail because truffleruby `must_equal` check the exact precision of float point
```

Comment:
Truffleruby will compare the exact float point when use `must_equal`.

For example, for scale =-10, boundary is `5.562684646268003e-309`; correct_boundary is `1/179769313486231590772930519078902473361797697894230657273430081157732675805500963132708477322407536021120113879871393357658789768814416622492847430639474124377767893424865485276302219601246094119453082952085005768838150682342462881473913110540827237163350510684586298239947245938479716304835356329624224137216`,

Since `5.562684646268003e-309` will assume all number after 3 will be 0, so it's different compare to correct_boundary. Truffleruby will treat them as not equal but other ruby variation will accept it.

Let me know if I miss something.